test/rules/avoid_renaming_method_parameters.dart

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2022, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:test_reflective_loader/test_reflective_loader.dart';
+
+import '../rule_test_support.dart';
+
+main() {
+  defineReflectiveSuite(() {
+    defineReflectiveTests(AvoidRenamingMethodParametersTest);
+  });
+}
+
+@reflectiveTest
+class AvoidRenamingMethodParametersTest extends LintRuleTest {
+  @override
+  List<String> get experiments => [
+        EnableString.enhanced_enums,
+      ];
+
+  @override
+  String get lintRule => 'avoid_renaming_method_parameters';
+
+  @FailingTest(issue: 'https://github.com/dart-lang/linter/issues/3095')
+  test_rename() async {
+    await assertDiagnostics(r'''
+class C {
+  int f(int f) => f;
+}
+enum A extends C {
```

Comment:
The [enhanced enum spec](https://github.com/dart-lang/language/blob/master/working/0158%20-%20Enhanced%20Enum/proposal.md) allows an enum to implement an interface, but doesn't allow it to extend another class (because they all implicitly extend `Enum`). So this test case is invalid.

Reply:
Aha!  Thanks for that clarification.  Updating to use `implements`.